src/node-transformations.js

```diff
@@ -0,0 +1,21 @@
+import markdownConverter from './markdown-converter';
+
+export const addSrcToImage = (node, pluginOptions) => {
+  if (node.internal.type !== 'CmsImage') return node;
+
+  return {
+    ...node,
+    src: `${pluginOptions.endpoint}${node.url}`,
+  };
+};
+
+export const addHtmlToTextBlock = node => {
+  if (node.internal.type !== 'CmsComponentText') return node;
+
+  return {
+    ...node,
+    html: markdownConverter.render(node.content),
+  };
+};
+
+export default [addSrcToImage, addHtmlToTextBlock];
```

Comment:
👍  nice idea having a transformation api :) 

Theoretically we could have a transformation generator function: 
```
addNodePropeties = (type, func)  => { 
   (node, pluginOptions) => {
     if (node.internal.type !== 'CmsImage') return node;
     attributes = func(node, pluginOptions)
     return { ...node, ...attributes}
  }
}

export const addSrcToImage = 
  addNodePropeties('CmsImage', (node, pluginOptions) => {src:  `${pluginOptions.endpoint}${node.url}`});

export const addHtmlToTextBlock = 
  addNodePropeties('CmsComponentText', node => { html: markdownConverter.render(node.content) });

export default [addSrcToImage, addHtmlToTextBlock];
```